WooCommerce/Classes/Model/Coupon+Woo.swift

```diff
@@ -42,7 +42,8 @@ extension Coupon {
         var calendar = Calendar.current
         calendar.timeZone = gmtTimeZone
 
-        let result = calendar.compare(expiryDate, to: now, toGranularity: .day)
+        // Compare the dates by hour to get around edge cases of timezone differences.
+        let result = calendar.compare(expiryDate, to: now, toGranularity: .hour)
```

Comment:
I believe that to include all the cases, we should compare by `minute` granularity because there are some time zones where there is also the half-hour or 45 minutes. For example, in India, the India Standard Time is +5:30, or in New Zealand, the time is UTC +12:45 / +13:45. WDYT?